### PR TITLE
Unify version string management in code

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
- Read `main.Version` from `misc/manifest.xml`, this way we only have
  one place to bump the version number before tagging. Also used
  by Extension Deployment Pipeline.
- Let `govvv` handle GitState/GitCommit/BuildDate etc.

cc: @paulmey since @boumenot is OOF.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ install:
   - sudo npm install -g npm
   - sudo npm install -g azure-cli
   - go get -u github.com/golang/lint/golint
+  - go get -u github.com/ahmetalpbalkan/govvv
```
Comment (paulmey) [inline .travis.yml]: shouldn't you also update the actual build script to use `govvv` instead of just `go`?

File: main/version.go [hunk 2]
```diff
@@ -5,33 +5,24 @@ import (
...
+
 }
 
 // DetailedVersionString returns a detailed version string including version
 // number, git commit, build date, source tree state and the go runtime version.
 func DetailedVersionString() string {
-	state := State
-	if state == "" {
-		state = "clean"
-	}
```
Comment (paulmey) [inline main/version.go]: Test failed: [TestDetailedVersionString](https://travis-ci.org/Azure/custom-script-extension-linux/builds/151601635#L1089)

**Comment by paulmey** [comment]
@ahmetalpbalkan looks like this needs some more fixin'

File: .travis.yml (hunk 1 above)
Comment (ahmetalpbalkan) [inline .travis.yml]: @paulmey Makefile below uses `govvv`

**Comment by ahmetalpbalkan** [comment]
@paulmey pushed the test fix

File: .travis.yml (hunk 1 above)
Comment (paulmey) [inline .travis.yml]: Ah yes, completely missed that. Sorry.

**Comment by paulmey** [comment]
LGTM

**Pull request merged**